webapp/views/App/views/Data/Charts/components/Chart/components/ChartTypes/PieChart/utils/processData.js

```diff
@@ -0,0 +1,73 @@
+export const processData = (originalData, specs) => {
+  const groupByField = specs?.query?.groupBy?.field
+  const metricField = specs?.query?.metric?.field
+  const aggregateType = specs?.query?.metric?.aggregate
+
+  let aggregatedData = {}
+
+  // Process each item in the chart result
+  originalData?.chartResult?.forEach((item) => {
+    const categoryValue = item[groupByField] || 'No Category'
+    const metricValue = parseFloat(item[metricField])
+
+    if (!isNaN(metricValue)) {
+      if (!aggregatedData[categoryValue]) {
```

Comment:
aggregatedData[categoryValue] is used several times above, can it be extracted into a constant?